README.md

```diff
@@ -0,0 +1,217 @@
+| `MIGRATION_FAILED`          | The migration failed.                                      |
+| `VERIFYING`                 | The migration is verifying that the migration was correct. |
+| `VERIFICATION_FAILED`       | `VERIFYING` failed.                                        |
+| `MIGRATION_SUCCEEDED`       | The migration succeeded.                                   |
+
+There are additional states for migrations that support reverting migrations:
+
+| State                        | Description                              |
+|------------------------------|------------------------------------------|
+| `REVERT_TESTING`             | The initial stage when running a revert. |
+| `REVERT_TESTING_FAILED`      | `REVERT_TESTING` failed.                 |
+| `REVERT_MIGRATING`           | Revert is being executed.                |
+| `REVERT_MIGRATION_FAILED`    | `REVERT_MIGRATING` failed.               |
+| `REVERT_VERIFYING`           | Verifying that a revert succeeded.       |
+| `REVERT_VERIFICATION_FAILED` | The revert failed.                       |
+| `REVERT_SUCCEEDED`           | The revert succeeded.                    |
+
+Reverting a migration is useful for quick rollback of a failed. Also, each
+state is also accompanied by a free-text `payload` containing additional
+information.
+
+The following state transitions exist:
+
+| Name | Source states | Destination state |
+|------|---------------|-------------------|
+| migrate | `NEW` | `TESTING` |
+| fail-testing | `TESTING` | `TESTING_FAILED` |
+| execute-migration | `TESTING` | `MIGRATING` |
+| fail-migration | `MIGRATING` | `MIGRATION_FAILED` |
+| verify-migration | `MIGRATING` | `VERIFYING`
+| fail-verification | `VERIFYING` | `VERIFICATION_FAILED` |
+| succeed-migration | `VERIFYING` | `MIGRATION_SUCCEEDED` |
+| retry-migration | `TESTING_FAILED`, `MIGRATION_FAILED`, `VERIFICATION_FAILED` | `TESTING` |
+| revert-migration | `MIGRATION_SUCCEEDED` | `REVERT_TESTING` |
+| fail-revert-testing | `REVERT_TESTING` | `REVERT_TESTING_FAILED` |
+| execute-revert | `REVERT_TESTING` | `REVERT_MIGRATING` |
+| fail-revert-execution | `REVERT_MIGRATING` | `REVERT_MIGRATION_FAILED` |
+| verify-revert | `REVERT_MIGRATING` | `REVERT_VERIFYING` |
+| fail-revert-verification | `REVERT_VERIFYING` | `REVERT_VERIFICATION_FAILED` |
+| succeed-migraton | `REVERT_VERIFYING` | `REVERT_SUCCEEDED` |
+| retry-revert | `REVERT_TESTING_FAILED`, `REVERT_MIGRATION_FAILED`, `REVERT_VERIFICATION_FAILED` | `REVERT_TESTING` |
+
+Getting started
+---------------
+
+First, [install Go](https://golang.org/dl/) and then set the `GOPATH`
+environment variable to a directory where you'd like to install the binaries.
+You can then install the binaries needed for this example by issuing
+
+```bash
+$ go install github.com/any-migrate/any-migrate
+$ go install github.com/any-migrate/file-repository
+$ go install github.com/any-migrate/mysql-migrator
```

Comment:
I think splitting this into separate repositories probably isn't a good idea in the beginning. Better to stabilize integration points (that is, interfaces) before splitting out into separate repositories.